packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -876,9 +876,9 @@ function readCodeCacheFile(cachePath: Config.Path): string | null {
   if (content == null) {
     return null;
   }
-  const code = content.substr(33);
+  const code = content.substring(33);
   const checksum = createHash('md5').update(code).digest('hex');
-  if (checksum === content.substr(0, 32)) {
+  if (checksum === content.substring(0, 32)) {
```

Comment:
this is a change in logic, no? IIRC of one `substr` and `substring` is inclusive on the second argument, while the other is not

Reply:
yes  
in substr the second agr. is the max length of string to return
and in substring it is the index to stop at but not included
this will  change logic if first argument is not 0
 but here the first argument is 0 and it is hardcoded ,I think it should work fine 🙂

Reply:
right! 👍